src/server.rs

```diff
@@ -55,7 +57,22 @@ async fn create_server() -> axum::Server<AddrIncoming, IntoMakeService<axum::Rou
         )
         .with_state(conf.leptos_options.clone());
 
-    let server = axum::Server::bind(&conf.leptos_options.site_addr).serve(app.into_make_service());
+    let site_addr = leptos_site_addr
+        .map(|s| s.parse::<SocketAddr>())
+        .map(|s| match s {
+            Ok(site_addr) => site_addr,
+            Err(err) => {
+                tracing::error!(
+                    "{}, defaulting to: {}",
```

Comment:
If the user passes bad address, we should fail right away, without falling back. Using `Option<String>` in the Args type should do it automatically.